Issue #17571: Update the documentation of AnnotationLocation

Tieshiyang opened the pull request:
Issue https://github.com/checkstyle/checkstyle/issues/17571
Clarified the documentation in `AnnotationLocationCheck.java` to indicate that the check does not enforce annotations to appear immediately after Javadocs, and points users to `InvalidJavadocPosition` for that purpose.
Appreciate the guidance provided in this issue.

Tieshiyang commented:
Thank you for pointing it out!
I've updated the line to comply with the line length limit and verified the build locally using `mvn clean verify`.
I also noticed that the tests only pass if all generated documentation `.xml` files are committed, so I’ve included them as well.

romani commented:
GitHub, generate website

romani commented:
GitHub, generate website

Tieshiyang commented:
I noticed that the Note section from `AnnotationLocationCheck.java` didn’t appear in the generated webpage. I modeled it after the `Note` in `IllegalTypeCheck.java`, so I’m curious why it wasn’t included. Just wondering if there’s a specific reason it’s not showing up, or if I should adjust something. Thanks in advance!

romani commented on src/site/xdoc/checks/annotation/annotationlocation.xml:
```diff
@@ -12,10 +12,10 @@
       <subsection name="Description" id="Description">
         <div>
           Checks location of annotation on language elements.
-          By default, Check enforce to locate annotations immediately after
```
Please look at IDE file that is  sibling to this file, it is named template.
You need to add macros to template.
Looks at example commit that I referenced.

romani reviewed (changes requested):
Items

romani commented:
https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=31545&view=logs&jobId=69ff11bc-5a77-53a4-545f-8dfc2c866a0e this means you are too behind in your forked master branch sync.
It is not blocker for PR but on next code change, please rebase to most latest.

romani commented:
@SteLeo1602 , please help here properly use Note macros in template file

Tieshiyang commented:
Sorry for the mistake in my previous commit—I missed pushing the template file and updated xml file. I’ve now updated it, rebased, and pushed the changes. I really appreciate your guidance and patience throughout this process!

romani commented:
GitHub, generate website

romani reviewed (approved):
thanks a lot

romani commented:
website will be updated on next release
